src/libstore/lock.cc

```diff
@@ -7,6 +7,31 @@
 
 namespace nix {
 
+#if __linux__
+
+static std::vector<gid_t> get_group_list(const char *username, gid_t group_id)
+{
+    std::vector<gid_t> gids;
+    gids.resize(32); // Initial guess
+
+    auto getgroupl {[&] {
+        int ngroups = gids.size();
+        int err = getgrouplist(username, group_id, gids.data(), &ngroups);
+        gids.resize(ngroups);
+        return err;
+    }};
+
+    // The first -1 means that the vector was not big enough.
+    // If it happens again, there is some different problem.
+    if (-1 == getgroupl() && -1 == getgroupl()) {
+        throw Error("failed to get list of supplementary groups for '%s'", username);
```

Comment:
```suggestion
        throw SysError("failed to get list of supplementary groups for '%s'", username);
```
so we get some errno and friends info.